liota/core/package_manager.py

```diff
@@ -376,37 +359,67 @@ def run(self):
 
             # Switch on message content (command), determine what to do
             command = msg[0]
-            if command in ["load", "unload", "delete", "reload", "update"]:
+            if command in ["load", "reload", "update"]:
                 #-----------------------------------------------------------
                 # Use these commands to handle package management tasks
 
                 with package_lock:
                     if len(msg) < 2:
                         log.warning("No package is specified: %s" % command)
                         continue
-                    if len(msg) > 2:
-                        list_packages = msg[1:]
+                    if len(msg) > 3:
+                        list_arg = msg[1:]
+                        list_packages = []
+                        cnt = len(list_arg)
+                        i = 0
+                        while (i < cnt):
+                            list_packages.append({list_arg[i]: list_arg[i+1]})
```

Comment:
I don't think it will even reach checksum calculations. In case of odd number of arguments, accessing list_arg[i+1] will give an index out of range error. I think we should add a check to check the number of arguments before accessing (i+1)th element of the list.